### DISCUSSION TITLE
disable docblock_missing_param Report

### DISCUSSION THREAD

**Discussion by AntistressStore** [phpactor/phpactor#2545] [General]
Hello. Please tell me, is it possible to somehow **disable** specific diagnostic alerts? https://phpactor.readthedocs.io/en/master/reference/diagnostic.html
For example, through settings https://phpactor.readthedocs.io/en/master/reference/configuration.html. 

I'm especially interested in **"docblock_missing_param"**.
My IDE nvim

**Comment by dantleech** [comment]
No, it's not currently possible but it's on the imaginary TODO list 🙂

**Comment by downhiller** [comment]
Is this still not possible?  This pops up everytime I have a function param typed as "array".  Sure, I'd rather specify "array[int]", but we obviously can't, and I'd rather not add a docblock for every instance.  I am a team of one, I don't need to worry about standards for other people.

**Comment by dantleech** [comment]
yes you can now disable them:

https://phpactor.readthedocs.io/en/master/reference/configuration.html#language-server-diagnostic-ignore-codes

the codes are provided via. LSP...

**Comment by downhiller** [comment]
Provided where?  In my editor on the offending line I get something like `Method "__construct" is missing @param $options"`

My best punt from that is that if that's the info you're referring to, I'd have to guess that the relevant code is "method_is_missing_param", but adding this line:

`"language_server.diagnostic_ignore_codes": ["method_is_missing_param"]`

to my `.phpactor.json` file didn't make any difference.

If the LSP is providing a code elsewhere, I've no idea where - nothing useful in `:LspLog` or `:LspInfo`?

Thanks Dan, I appreciate your help.

**Reply by dantleech** [reply]
f.e. I see the code `worse.missing_member`:
![image](https://github.com/user-attachments/assets/ba07019c-d803-4029-bae1-0f6511bbf5f8)
which I see when I invoke:

```
vim.api.nvim_set_keymap('n', ',di', '<cmd>lua vim.diagnostic.open_float()<CR>', opts)
```
on the line with the diagnostic.

for the one your interested in I think it's `worse.docblock_missing_param`

**Comment by downhiller** [comment]
Thank you.  `worse.docblock_missing_param` was what I needed 👍

(Which I see now you said, I'm not good at reading and implemented the keymap to test for myself before I got that far 😁)
